docs/CONTRIBUTING.md

```diff
@@ -3,199 +3,208 @@ id: contributing
+    - Do not capitalize first character
+    - Do not end with a period
+    - Use imperative mood, in present tense; commit title should always be able to complete the following sentence:
+        - If applied, this commit will _commit title here_
+- body:
+    - Separate from subject with a blank line
+    - Wrap texts at 72 characters
+    - Explain _what_ and _why_, not _how_
+    - [GitHub flavored Markdown](https://github.github.com/gfm/) is ok to use
+    - Start with `BREAKING CHANGE: ` when you made significant change in the commit.
+
+Example commit message:
 
-```sh
-yarn test
 ```
+test(textlint-formatter): check types while testing
 
-You can run unit test in each package.
+- Add strict type check option to `ts-node` to make sure future
+  interface changes will be took into account while running test.
+- Update test case for interface changes made at #430.
 
-```sh
-cd pacakges/<pacakge>
-yarn test
+Closes #448.
 ```
 
-#### Example Test
+Please see [Commit Message Format](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular), [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/), and [Semantic Commit Messages](https://seesparkbox.com/foundry/semantic_commit_messages) for detail.
 
-Test [examples/*](../examples/) as test.
+##### Versioning
 
-```sh
-yarn run test:examples
-```
+We care version number when releasing so you should not modify `version` field of `package.json`. For the record, we use [Semantic Versioning](https://semver.org/).
 
-#### Integration Test
+- Patch release (intended to not break your lint build)
+    - A bug fix to the CLI or core (including formatters)
+    - Improvements to documentation
+    - Non-user-facing changes such as refactoring
+    - Re-releasing after a failed release (i.e., publishing a release that doesn't work for anyone)
+- Minor release (might break your lint build)
+    - A new option
+    - An existing rule is deprecated
+    - A new CLI capability is created
+    - New public API are added (new classes, new methods, new arguments to existing methods, etc.)
+        - It might break type interface(`.d.ts`)
+    - A new formatter is created
+- Major release (break your lint build)
+    - A new option to an existing rule that results in ESLint reporting more errors by default
+    - An existing formatter is removed
+    - Part of the public API is removed or changed in an incompatible way
 
-Run [Integration test](../test/integration-test) that use actual textlint user project.
-This test is heavy.
+### Running Tests
 
-```sh
-# Require git submodule
-git submodule update --init
-# Run test
-yarn run test:integration
-```
+We have four type of tests. You should run at least **unit test** or **documentation test** according to your type of changes before submitting a pull request.
 
-#### Documentation Test
+1. Unit test: `yarn test` to run tests under [`packages/`](../packages)
```

Comment:
Each line has mixed three things.

- Category
- What is target?(Description)
- How to run test?(Command)

To separate things, we can get copy-pastability and readablity.

- Category
  - Description
  - Command

```markdown
### Category

> Description

   $ Command
```

A contributor is not always see all paragraph. They may see only Header text.
Mixing description and command in a paragraph make them confuse.
The tendency is particularly strong among non-native speaker.
(My opinion has been affected by Paragraph Reading)

I think it's good that separate description and command.
Probably, the text will be long in compensation for separating.
It is a trade-off.

Reply:
Current: 

----

### Running Tests 

We have four type of tests. You should run at least **unit test** or **documentation test** according to your type of changes before submitting a pull request.

1. Unit test: `yarn test` to run tests under [`packages/`](../packages)
2. Example test: `yarn test:examples` to run [`examples`](../examples) as test
3. Integration test: `git submodule update --init && yarn test:integration` to run tests with [real–world documents](../test/integration-test)
4. Documentation test: `yarn test:docs` to run textlint to [`docs`](../docs), [`.github`](../.github), and [README](../README.md) (we are dog–fooding!)

----

Reply:
My proposal:

----

### Running Tests 

We have four type of tests. You should run at least **unit test** or **documentation test** according to your type of changes before submitting a pull request.

#### Unit test

Run tests under [`packages/`](../packages).

```sh
$ yarn test
```

#### Example test

Run [`examples`](../examples) as test,

```sh
$ yarn test:examples
```

#### Documentation test

Run tests with [real–world documents](../test/integration-test)

```sh
$ git submodule update --init && yarn test:integration
```

#### Integration test

Run textlint to [`docs`](../docs), [`.github`](../.github), and [README](../README.md) (We are dog–fooding!).

```sh
$ yarn test:docs
```

----

Reply:
@azu, thank you for your review. I totally agree with you as non–native speaker :)

I'll add another commit according to your proposal.